src/broker/publisher.c

```diff
@@ -54,13 +53,12 @@ static flux_msg_t *encode_event (const char *topic, int flags,
     }
     if (src) { // optional payload
         int srclen = strlen (src);
-        size_t dstlen = BASE64_DECODE_SIZE (srclen);
+        size_t dstbuflen = base64_decoded_length (srclen);
+        ssize_t dstlen;
 
-        if (!(dst = malloc (dstlen)))
+        if (!(dst = calloc (1, dstbuflen)))
             goto error;
-        if (sodium_base642bin ((unsigned char *)dst, dstlen, src, srclen,
-                               NULL, &dstlen, NULL,
-                               sodium_base64_VARIANT_ORIGINAL) < 0) {
+        if ((dstlen = base64_decode (dst, dstbuflen, src, srclen)) < 0) {
```

Comment:
I'm not sure there is any purpose served by zero-filling the buffer passed to `base64_decode()`.  Use `malloc()`?